### ISSUE TITLE
if you want to use gradient penalty, you have to build from source?

### ISSUE THREAD

**Issue by dragen1860** [aelnouby/Text-to-Image-Synthesis#13]
Hi, I want to use gradient penalty since i wana try improved wgan.
what does `build from source ` mean? 
it means I should train it from scratch or build pytorch souce code?

thanks.

**Comment by aelnouby** [comment]
It means build pytorch from source code.

**Comment by dragen1860** [comment]
Hi, why should build pytorch from source code? Could you explain that? thanks.

**Comment by aelnouby** [comment]
The pytorch version you install using pip or conda is not the latest, it is the latest stable version they have published, sometimes they implement features on the repo but they are not yet added to the published version, in this case you need the build their source code from the master branch to get the latest features, second order gradients follows this case.

However, last week pytorch published a new version, so it might include the second order gradients. In this case you do not need to build from source, however I am not sure if it is included or not.

**Comment by dragen1860** [comment]
Hi, Pytorch support higher order gradients from v0.2 already, please refer to : https://github.com/pytorch/pytorch/releases/tag/v0.2.0

**Comment by aelnouby** [comment]
Great ! at the time of this project development this feature was not available. I will edit the ReadMe. Thanks.
